### PR TITLE
docs for installing software via an easystack file

### PR DESCRIPTION
**Pull request by deniskristak**
docs for Building with easystack

https://github.com/easybuilders/easybuild-framework/pull/3479

### CODE REVIEW

File: docs/Building_with_easystack.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    R:
+      toolchains:
+        foss-2020a:
+          versions: [4.0.0]
+
+Installation using this *easystack* called 'myeasystack.yaml' with command
+
+.. code::
+
+  eb --easystack myeasystack.yaml;
```
Comment (Darkless012) [inline docs/Building_with_easystack.rst:104]: why the semicolon `;`?

File: docs/Building_with_easystack.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+.. code::
+
+  eb Bioconductor-3.11-foss-2020a.eb;
+  eb EasyBuild-4.3.1-SYSTEM.eb;
+  eb GROMACS-2020.1-foss-2020a.eb;
+  eb GROMACS-2020.3-foss-2020a.eb;
+  eb GROMACS-2020.1-fosscuda-2020a.eb;
+  eb OpenFOAM-8-foss-2020a.eb;
+  eb OpenFOAM-v2006-foss-2020a.eb;
+  eb R-4.0.0-foss-2020a.eb;
```
Comment (Darkless012) [inline docs/Building_with_easystack.rst:117]: I believe it is possible to run all the easyconfigs in one eb command, which is what I think easystack actually currenlty does?
```eb Bioconductor-3.11-foss-2020a.eb EasyBuild-4.3.1-SYSTEM.eb GROMACS-2020.1-foss-2020a.eb  GROMACS-2020.3-foss-2020a.eb GROMACS-2020.1-fosscuda-2020a.eb OpenFOAM-8-foss-2020a.eb OpenFOAM-v2006-foss-2020a.eb R-4.0.0-foss-2020a.eb
```

File: docs/Building_with_easystack.rst (hunk 1 above)
Comment (deniskristak) [reply docs/Building_with_easystack.rst]: removed :-)

File: docs/Building_with_easystack.rst (hunk 2 above)
Comment (deniskristak) [reply docs/Building_with_easystack.rst]: that is true, changed

File: docs/Building_with_easystack.rst [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+.. building_with_easystack_usage:
+
+Usage
+-------------------------
+
+To build software with *Easystack*, type:
+
+.. code::
+
+  eb --easystack yourEasystackFile.yaml
```
Comment (boegel) [inline docs/Building_with_easystack.rst:33]: use `example.yaml`

File: docs/Building_with_easystack.rst [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+where yourEasystackFile.yaml is the file with specifications that you just created (more about this in the next chapter).
+
+.. building_with_easystack_structure:
+
+Structure of an Easystack file
+-------------------------
+
+Easystack file can currently be of type YAML.
```
Comment (boegel) [inline docs/Building_with_easystack.rst:42]:
```
Easystack files are written in YAML syntax.
```

Is there a good guide to YAML syntax you can link to?

File: docs/Building_with_easystack.rst [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+*versions*: Version of software. If multiple entries are provided, EB will install all of them. 
+Can be in form of a list or consecutive line entries (see example). 
+
+**General structure of YAML-formatted *easystack*:**
+
+.. code::
+
+  software:
+    software_name:
+      toolchain:
```
Comment (boegel) [inline docs/Building_with_easystack.rst:64]: should be `toolchains:`

File: docs/Building_with_easystack.rst [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+  eb --easystack myeasystack.yaml
+
+is an equivalent of running:
+
+.. code::
+
+  eb Bioconductor-3.11-foss-2020a.eb EasyBuild-4.3.1-SYSTEM.eb GROMACS-2020.1-foss-2020a.eb GROMACS-2020.3-foss-2020a.eb GROMACS-2020.1-fosscuda-2020a.eb OpenFOAM-8-foss-2020a.eb OpenFOAM-v2006-foss-2020a.eb R-4.0.0-foss-2020a.eb
+
+to be developed:
```
Comment (boegel) [inline docs/Building_with_easystack.rst:112]: Make this a subsection:

```
To be developed
---------------
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**